sechub-scan/src/test/java/com/mercedesbenz/sechub/domain/scan/asset/AssetServiceTest.java

```diff
@@ -252,6 +254,34 @@ void ensureAssetFileInStorageAvailableAndHasSameChecksumAsInDatabase_in_storage_
 
     }
 
+    @Test
+    void delete_non_existing_single_asset_file_throws_exception() throws IOException {
+        /* prepare */
+        String assetId = "asset1";
+        String fileName = "file1.txt";
+
+        AssetStorage assetStorage = mock();
+        when(storageService.createAssetStorage(assetId)).thenReturn(assetStorage);
+        doNothing().when(assetStorage).delete(fileName);
+
+        /* execute + test */
+        assertThatThrownBy(() -> serviceToTest.deleteAssetFile(assetId, fileName)).isInstanceOf(NotFoundException.class).hasMessageContaining(assetId)
+                .hasMessageContaining(fileName);
+    }
+
+    @Test
+    void delete_non_existing_asset_throws_exception() throws IOException {
+        /* prepare */
+        String assetId = "asset1";
+
+        AssetStorage assetStorage = mock();
+        when(storageService.createAssetStorage(assetId)).thenReturn(assetStorage);
+        doNothing().when(assetStorage).deleteAll();
```

Comment:
See my comments before about the doNoting() part (which is IMHO useless/irritating here) and the missing job repository setup for the delete call